src/base/i18n.cc

```diff
@@ -141,250 +257,40 @@ AddOnTextdomain::AddOnTextdomain(std::string addon, const int i18n_version) {
-	   {"hu", "hu,hu_HU"},
-	   {"ia", "ia"},
-	   {"id", "id,id_ID"},
-	   {"it", "it,it_IT,it_CH"},
-	   {"ja", "ja,ja_JP"},
-	   {"jv", "jv,jv_ID,jv_MY,jv_SR,jv_NC"},
-	   {"ka", "ka,ka_GE"},
-	   {"ko", "ko,ko_KR"},
-	   {"la", "la,la_AU,la_VA"},
-	   {"mr", "mr,mr_IN"},
-	   {"ms", "ms,ms_MY"},
-	   {"my", "my,my_MM"},
-	   {"nb", "nb,nb_NO"},
-	   {"nl", "nl,nl_NL,nl_BE,nl_AW"},
-	   {"nn", "nn,nn_NO"},
-	   {"oc", "oc,oc_FR"},
-	   {"pl", "pl,pl_PL"},
-	   {"pt", "pt,pt_PT,pt_BR"},
-	   {"pt_BR", "pt_BR,pt,pt_PT"},
-	   {"ru", "ru,ru_RU,ru_UA"},
-	   {"si", "si,si_LK"},
-	   {"sk", "sk,sk_SK"},
-	   {"sl", "sl,sl_SI"},
-	   {"sr", "sr,sr_RS,sr_ME"},
-	   {"sv", "sv,sv_FI,sv_SE"},
-	   {"tr", "tr,tr_TR,tr_CY"},
-	   {"uk", "uk,uk_UA"},
-	   {"vi", "vi,vi_VN"},
-	   {"zh", "zh,zh_CN,zh_TW,zh_HK,zh_MO"},
-	   {"zh_CN", "zh_CN,zh,zh_TW,zh_HK,zh_MO"},
-	   {"zh_TW", "zh_TW,zh_HK,zh_MO,zh,zh_CN"},
-	};
-
-	std::string lang(name);
-
-	log_info("selected language: %s\n", lang.empty() ? "(system language)" : lang.c_str());
-
-#ifndef _WIN32
-#ifndef __AMIGAOS4__
-#ifndef __APPLE__
-	unsetenv("LANGUAGE");  // avoid problems with this variable
-#endif
-#endif
-#endif
-
-	std::string alt_str;
-	if (lang.empty()) {
-		// reload system language, if selected
-		lang = env_locale;
-		alt_str = env_locale;
-	} else {
-		alt_str = lang;
-		// otherwise, try alternatives.
-		if (kAlternatives.count(lang) != 0u) {
-			alt_str = kAlternatives.at(lang);
-		}
-	}
-	alt_str += ",";
-
-// Somehow setlocale doesn't behave same on
-// some systems.
-#ifdef __BEOS__
-	setenv("LANG", lang.c_str(), 1);
-	setenv("LC_ALL", lang.c_str(), 1);
-	locale = lang;
-#endif
-#ifdef __APPLE__
-	setenv("LANGUAGE", lang.c_str(), 1);
-	setenv("LANG", lang.c_str(), 1);
-	setenv("LC_ALL", lang.c_str(), 1);
-	locale = lang;
-#endif
-#ifdef _WIN32
-	_putenv_s("LANG", lang.c_str());
-	locale = lang;
-#endif
-
-#ifdef __linux__
-	char* res = nullptr;
-	char const* encodings[] = {"", ".utf-8", "@euro", ".UTF-8"};
-	std::size_t found = alt_str.find(',', 0);
-	bool leave_while = false;
-	// try every possible combination of alternative and encoding
-	while (found != std::string::npos) {
-		std::string base_locale = alt_str.substr(0, int(found));
-		alt_str = alt_str.erase(0, int(found) + 1);
-
-		for (char const* encoding : encodings) {
-			std::string try_locale = base_locale + encoding;
-			res = SETLOCALE(LC_MESSAGES, try_locale.c_str());
-			if (res != nullptr) {
-				locale = try_locale;
-				log_info("using locale %s\n", try_locale.c_str());
-				leave_while = true;
-				break;
-			}  // log("locale is not working: %s\n", try_locale.c_str());
-		}
-		if (leave_while) {
+void init_locale() {
+	tinygettext::Log::set_log_error_callback(tinygettext_log_err);
+	tinygettext::Log::set_log_warning_callback(tinygettext_log_warn);
+	tinygettext::Log::set_log_info_callback(tinygettext_log_info);
+
+	locale = std::string();
+	env_locale = "en";
+	for (const auto& var : {"LANG", "LANGUAGE"}) {
```

Comment:
Shouldn't we also try `LC_ALL` and `LC_MESSAGES`? Also, the precedence should be `LANGUAGE` (may contain multiple entries also in order of preference... but I don't think we can do that properly, ie. fall back per string...), `LC_ALL`, `LC_MESSAGES`, `LANG`.

Or is your problem that your system messes with those in an uncontrollable way?